pkg/controller/operconfig/operconfig_controller.go

```diff
@@ -176,7 +176,30 @@ func (r *ReconcileOperConfig) Reconcile(request reconcile.Request) (reconcile.Re
 	// is safe.
 	if prev != nil {
 		// Check if the operator is put in the 'Network Migration' mode.
-		if _, ok := operConfig.GetAnnotations()[names.NetworkMigrationAnnotation]; !ok {
+		if val, ok := operConfig.GetAnnotations()[names.NetworkMigrationAnnotation]; !ok || val != "start" {
```

Comment:
How does this work? If the annotation doesn't exist how do you know it is in network migration mode? To me it would seem the opposite would be true. Should this be

```
if val, ok := operConfig.GetAnnotations()[names.NetworkMigrationAnnotation]; ok && val != "start" {
```

Reply:
The condition here is to check if CNO shall bypass `IsChangeSafe`. So if the annotation doesn't exist or it exists but its value is not 'start', both mean CNO shall not bypass `IsChangeSafe` check and start the migration.

If the annotation exists and its value is 'prepare', the CNO comes to the migration preparing phase.

Reply:
OK I don't think I understand CNO code well enough to review this. @abhat do you mind taking a look?